src/lib/ext-popup.js

```diff
@@ -0,0 +1,93 @@
+import {
+  areExtsBeingMonitored,
+  initMonitorAll,
+  stopMonitorAll,
+  viewActivityLogs,
+} from '../lib/ext-listen.js';
+
+export default class Popup {
+  renderErrorMsg(message) {
+    this.errorMsgText.textContent = message;
+    this.errorMsgText.style.display = 'block';
+  }
+
+  renderMonitorStartedUI() {
+    this.startMonitorAllBtn.setAttribute('disabled', 'disabled');
+    this.startMonitorAllBtn.classList.add('disabled');
+    this.stopMonitorAllBtn.removeAttribute('disabled');
+    this.stopMonitorAllBtn.classList.remove('disabled');
+
+    this.monitorStatusText.textContent = 'Extensions are being monitored';
+    this.monitorStatusText.classList.add('green');
+    this.monitorStatusText.classList.remove('red');
+  }
+
+  renderMonitorStoppedUI() {
+    this.stopMonitorAllBtn.setAttribute('disabled', 'disabled');
+    this.stopMonitorAllBtn.classList.add('disabled');
+    this.startMonitorAllBtn.removeAttribute('disabled');
+    this.startMonitorAllBtn.classList.remove('disabled');
+
+    this.monitorStatusText.textContent = 'No extensions are being monitored';
+    this.monitorStatusText.classList.add('red');
+    this.monitorStatusText.classList.remove('green');
+  }
+
+  async renderPopup() {
+    const isMonitorStarted = await areExtsBeingMonitored();
+    isMonitorStarted
+      ? this.renderMonitorStartedUI()
+      : this.renderMonitorStoppedUI();
+  }
+
+  addListenerPopup(element, event, listener) {
+    element.addEventListener(event, listener);
+  }
+
+  initializeMonitoring = async () => {
+    initMonitorAll()
+      .then((monitorMsg) => {
+        if (monitorMsg === 'ext-monitor-started') {
+          this.renderPopup();
+        }
+      })
+      .catch((error) => {
+        this.renderErrorMsg(error?.message);
+      });
+  };
```

Comment:
Yeah, you guessed right. Since I had made the workaround using an arrow function, I just ignored if I could do that in any other way :-p. Also, I didn't know about other workarounds and about the `handleEvent method`. Thanks a lot for the explanation. I have updated this part with your suggestion (added an handleEvent method and used try, catch)

> As a side not given that the reference to the button is called `startMonitorAllBtn` and the other method is called `stopMonitoring`, we could use the same convention can call the methods `startMonitor` and `stopMonitor`.

(I understand you wanted to write `note` instead of `not`). But I am not sure I understand this clearly. I have renamed the methods to `startMonitor` and `stopMonitor`.